OpenRA.Mods.Common/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -57,25 +57,26 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 			CPos endPos;
 			WAngle spawnFacing;
 
+			var bounds = map.Bounds;
+			var diagonal = Exts.ISqrt(bounds.Height * bounds.Height + bounds.Width * bounds.Width);
+
 			if (info.BaselineSpawn && mpStart != null)
 			{
 				var spawn = mpStart.Start[owner];
-				var bounds = map.Bounds;
 				var center = new MPos(bounds.Left + bounds.Width / 2, bounds.Top + bounds.Height / 2).ToCPos(map);
 				var spawnVec = spawn - center;
-				startPos = spawn + spawnVec * (Exts.ISqrt((bounds.Height * bounds.Height + bounds.Width * bounds.Width) / (4 * spawnVec.LengthSquared)));
+				startPos = spawn + spawnVec * diagonal / (2 * spawnVec.Length);
 				endPos = startPos;
-				var spawnDirection = new WVec((self.Location - startPos).X, (self.Location - startPos).Y, 0);
-				spawnFacing = spawnDirection.Yaw;
+				spawnFacing = map.FacingBetween(startPos, self.Location, WAngle.Zero);
 			}
 			else
 			{
-				// Start a fixed distance away: the width of the map.
+				// Start a fixed distance away: the diagonal of the map.
 				// This makes the production timing independent of spawnpoint
-				var loc = self.Location.ToMPos(map);
-				startPos = new MPos(loc.U + map.Bounds.Width, loc.V).ToCPos(map);
-				endPos = new MPos(map.Bounds.Left, loc.V).ToCPos(map);
 				spawnFacing = info.Facing;
+				var spawnVec = new MPos(spawnFacing.Sin() * diagonal / 1024, spawnFacing.Cos() * diagonal / 1024).ToCPos(map) - CPos.Zero;
```

Comment:
This is going to nerf vehicle delivery time in TD and other mods that don't use `BaselineSpawn` by 40%, with no return benefit. I suspect this won't be wanted after all the hassle that went into trying to decrease the spawn time.

Reply:
Fixed the math so the length of the approach is not longer than necessary given the specific fixed angle.